src/evolutionaryAlgorithm/SimulationComponents/ParentSelector/impl/rouletteParentSelector.py

```diff
@@ -0,0 +1,46 @@
+from typing import List
+from random import uniform
+
+from src.evolutionaryAlgorithm.SimulationComponents.ParentSelector.ParentSelectorInterface \
+    import ParentSelectorInterface
+from src.evolutionaryAlgorithm.SimulationComponents.Individual.IndividualInterface import IndividualInterface
+
+
+class rouletteParentSelector(ParentSelectorInterface):
+
+    def __init__(self):
+        pass
+
+    def marry(self, Population: List[IndividualInterface], fitness: List[float]):
+        popNum = len(Population)
+        max, min, sum = 0, fitness[0], 0
+        for i in range(popNum):
+            sum += fitness[i]
+            if min > fitness[i]:
+                min = fitness[i]
+            if max < fitness[i]:
+                max = fitness[i]
+
+        param = max - min
+        fitnessPrim = []
+        for i in range(popNum):         #q' = (q-min(q))*(max(q)-min(q))
+            a = fitness[i]-min
+            b = a*param                 #q" = 1-q'
+            fitnessPrim.append(1-b)
+
+        marriages = []
+        for i in range(int(popNum/2)):
+            parents = []
+            for j in range(2):
+                k = self.chooseOne(fitnessPrim, sum)
+                parents.append(Population[k])
+            marriages.append(parents)
+        return marriages
```

Comment:
Jest to długa metoda która implementuje logikę, przydałoby się zrobić test jesdnostkowy żeby sprawdzić czy działa dla chociaż prostych przypadków